pkg/operator/operator.go

```diff
@@ -214,6 +214,19 @@ func (k *Kontroller) Run(ctx context.Context) error {
 			continue
 		}
 
+		if manageAgent {
+			agentsReady, err := xyz()
+			if err != nil {
+				glog.Errorf("unable to ensure managed agents are ready: %v", err)
+				continue
+			}
+			if !agentsReady {
+				glog.V(4).Infof("waiting for managed agents to be ready, not doing additional work now")
+				// TODO: if this has happened for a long enough duration, it should become an error.
```

Comment:
This code was changed in a later version to poll via crashlooping if necessary (the agentsReady variable was entirely removed)